auto3dseg/algorithm_templates/dints/scripts/algo.py

```diff
@@ -388,8 +388,24 @@ def train(self, train_params=None):
         if train_params is not None:
             params = deepcopy(train_params)
 
+        # Overriding DiNTS parameters typically needs searching#<key> or training#<key>, but for API alignment, some
+        # keys must be allowed without "searching" and "training", such as 'num_epochs'. If a key can be found in
+        # searching in hyper_parameter_search.yaml or training in hyper_parameter.yaml, the key can be used directly
+        # without the prefix searching# or training#
+
+        output_path = self.fill_records["hyper_parameters.yaml"]["bundle_root"]
+
+        parser = ConfigParser(globals=False)
+        parser.read_config(os.path.join(output_path, "configs", "hyper_parameters_search.yaml"))
+
+        allow_search_set = [k for k in parser.get("searching")]
+
+        parser = ConfigParser(globals=False)
+        parser.read_config(os.path.join(output_path, "configs", "hyper_parameters.yaml"))
+        allow_train_set = [k for k in parser.get("training")]
+
         for k, v in params.items():
-            if k == "CUDA_VISIBLE_DEVICES":
+            if k not in allow_search_set:
```

Comment:
we can add additional warning messages here if the given keys are non-existed.

Reply:
It should be easy to add, but it means users will see a warning if they override parameters "cache_rate". Would logging at the info/debug level be sufficient, or should we still use a warning?

Reply:
@dongyang0122